src/pessimisticPatternMatcherUtils.cc

```diff
@@ -210,72 +196,65 @@ RotationTestResult test_rotation(ndarray::Array<double, 1, 1> const& src_center,
     }
     double sin_shift = sqrt(1 - cos_shift * cos_shift);
 
-    // If the sine of our shift is zero we only need to use the identity
-    // matrix for the shift. Else we construct the rotation matrix for
-    // shift.
+    // If the sine of our shift is zero we only need to use the identity matrix for the shift. Else we
+    // construct the rotation matrix for shift.
     auto ref_center_eigen = ndarray::asEigenMatrix(ref_center).head<3>();
-    Eigen::Matrix<double, 3, 3> shift_matrix;
+    Eigen::Matrix3d shift_matrix;
     if (sin_shift > 0) {
         Eigen::Vector3d rot_axis = ndarray::asEigenMatrix(src_center).head<3>().cross(ref_center_eigen);
         rot_axis /= sin_shift;
         shift_matrix = create_spherical_rotation_matrix(rot_axis, cos_shift, sin_shift);
     } else {
-        shift_matrix = Eigen::Matrix<double, 3, 3>::Identity();
+        shift_matrix = Eigen::Matrix3d::Identity();
     }
 
-    // Now that we have our shift we apply it to the src delta vector
-    // and check the rotation.
-    Eigen::Vector3d rot_src_delta = shift_matrix * ndarray::asEigenMatrix(src_delta).head<3>();
+    // Now that we have our shift we apply it to the src delta vector and check the rotation.
+    Eigen::Vector3d rot_src_delta = shift_matrix * ndarray::asEigenMatrix(src_delta);
     Eigen::Vector3d proj_src_delta = rot_src_delta - rot_src_delta.dot(ref_center_eigen) * ref_center_eigen;
-    Eigen::Vector3d ref_delta_eigen = ndarray::asEigenMatrix(ref_delta).head<3>();
+    Eigen::Vector3d ref_delta_eigen = ndarray::asEigenMatrix(ref_delta);
 
     Eigen::Vector3d proj_ref_delta =
             ref_delta_eigen - ref_delta_eigen.dot(ref_center_eigen) * ref_center_eigen;
     double proj_src_delta_sq = proj_src_delta.dot(proj_ref_delta);
     double cos_rot_sq = proj_src_delta_sq * proj_src_delta_sq /
                         (proj_src_delta.dot(proj_src_delta) * proj_ref_delta.dot(proj_ref_delta));
-    // If the rotation isn't in tolerance return an empty vector.
-    RotationTestResult result;
     if (cos_rot_sq < max_cos_rot_sq) {
-        return result;
+        // failed test
```

Comment:
I think the old comment was actually more informative...

Reply:
I've reworded it again: the old one was was misleading. The "empty vector" part isn't as important as the "success=false" part.